runtime/j9vm/javanextvmi.cpp

```diff
@@ -274,21 +274,56 @@ enterVThreadTransitionCritical(J9VMThread *currentThread, jobject thread)
 	MM_ObjectAccessBarrierAPI objectAccessBarrier = MM_ObjectAccessBarrierAPI(currentThread);
 	j9object_t threadObj = J9_JNI_UNWRAP_REFERENCE(thread);
 
-	while(!objectAccessBarrier.inlineMixedObjectCompareAndSwapU64(currentThread, threadObj, vm->virtualThreadInspectorCountOffset, 0, (U_64)-1)) {
+retry:
+	while (!objectAccessBarrier.inlineMixedObjectCompareAndSwapU64(currentThread, threadObj, vm->virtualThreadInspectorCountOffset, 0, (U_64)-1)) {
```

Comment:
Directly casting a negative number to an unsigned type is a common anti-pattern in OpenJ9: a better expression than `(U_64)-1` would be `~(U_64)0`.